### PR TITLE
Feature/tank level

### PR DESCRIPTION
**Pull request by Andileh**
Please review my pr

### CODE REVIEW

File: WaterLog_Backend/Controllers/TankLevelsController.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        public async Task<ActionResult<TankLevelsEntry>> Get(int id)
+        {
+            return await _db.TankLevels.FindAsync(id);
+        }
+
+        // POST api/values
+        [HttpPost]
+        public async Task Post([FromBody] TankLevelsEntry value)
+        {
+            await _db.TankLevels.AddAsync(value);
```
Comment (DreyerChris) [inline WaterLog_Backend/Controllers/TankLevelsController.cs]: You will probably want to check the ModelState here

File: WaterLog_Backend/Controllers/TankLevelsController.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            await _db.SaveChangesAsync();
+        }
+
+        // PUT api/values/5
+        [HttpPut("{id}")]
+        public async Task Put(int id, [FromBody] TankLevelsEntry value)
+        {
+            var old = await _db.TankLevels.FindAsync(id);
```
Comment (DreyerChris) [inline WaterLog_Backend/Controllers/TankLevelsController.cs]: Do error checking on this call, what happens if it was not found?

File: WaterLog_Backend/Controllers/TankLevelsController.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using Microsoft.Extensions.Configuration;
+using WaterLog_Backend.Models;
+
+namespace WaterLog_Backend.Controllers
+{
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class TankLevelsController : ControllerBase
+    {
```
Comment (DreyerChris) [inline WaterLog_Backend/Controllers/TankLevelsController.cs:19]: You should be returning action results on all of these functions i.e `Ok();`, `NotFound();`

File: WaterLog_Backend/Controllers/TankLevelsController.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            var old = await _db.TankLevels.FindAsync(id);
+            _db.Entry(old).CurrentValues.SetValues(value);
+            await _db.SaveChangesAsync();
+        }
+
+        // DELETE api/values/5
+        [HttpDelete("{id}")]
+        public async Task Delete(int id)
+        {
+            var entry = await _db.TankLevels.FindAsync(id);
```
Comment (DreyerChris) [inline WaterLog_Backend/Controllers/TankLevelsController.cs]: Do error checking on this call, what happens if it was not found?

File: WaterLog_Backend/Models/TankLevelsEntry.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WaterLog_Backend.Models
+{
+    public class TankLevelsEntry
+    {
+        [Key]
+        public int Tank_Id { get; set; }
```
Comment (DreyerChris) [inline WaterLog_Backend/Models/TankLevelsEntry.cs]: Remove the underscore, keep the property names consistent and PascalCase

File: WaterLog_Backend/Models/TankLevelsEntry.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        public int Percentage { get; set; }
+        public string Level_Status { get; set; }
```
Comment (DreyerChris) [inline WaterLog_Backend/Models/TankLevelsEntry.cs]: Same here

File: WaterLog_Backend/appsettings.json [hunk 7]
```diff
@@ -1,9 +1,13 @@
...
-    },
-    "AllowedHosts": "*"
+  "Logging": {
+    "LogLevel": {
+      "Default": "Warning"
+    }
+  },
+  "AllowedHosts": "*",
 
+  "LiveConnectionString": "Data Source=.\\SQLEXPRESS;Initial Catalog=iot;Integrated Security=true",
```
Comment (DreyerChris) [inline WaterLog_Backend/appsettings.json]: Henko told you guys you shouldn't be committing connection strings and passwords, Ivan will have to reset the passwords again

**Review by DreyerChris — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/appsettings.json [hunk 8]
```diff
@@ -1,9 +1,13 @@
...
+  "LocalLiveDBConnectionString": "Server=dev.retrotest.co.za;Database=iot;User Id=group1;Password=fNX^r+UKy3@CtYh5",
```
Comment (FredJEhlers) [inline WaterLog_Backend/appsettings.json]: No passwords in the repo please

File: WaterLog_Backend/appsettings.json (hunk 7 above)
Comment (Andileh) [reply WaterLog_Backend/appsettings.json]: i sent it by mistake am gonna take it out now

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 2 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: changes are done

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 1 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: I try to do it but am not wining please help

File: WaterLog_Backend/Models/TankLevelsEntry.cs (hunk 6 above)
Comment (Andileh) [reply WaterLog_Backend/Models/TankLevelsEntry.cs]: done

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 1 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: done

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 2 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: done

File: WaterLog_Backend/Models/TankLevelsEntry.cs (hunk 5 above)
Comment (Andileh) [reply WaterLog_Backend/Models/TankLevelsEntry.cs]: done

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 4 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs]: done

File: WaterLog_Backend/Controllers/TankLevelsController.cs (hunk 3 above)
Comment (Andileh) [reply WaterLog_Backend/Controllers/TankLevelsController.cs:19]: done

**Review by ash-rabbit — APPROVED** [review]

**Review by DreyerChris — APPROVED** [review]

**Review by RashadAkoodieRabbit — APPROVED** [review]

**Pull request merged**